### PR TITLE
ADEN-3551 Call revive in top button and wikiabar

### PR DESCRIPTION
**Pull request by drets**

### CODE REVIEW

File: extensions/wikia/AdEngine/js/SlotTweaker.js [hunk 1]
```diff
@@ -3,7 +3,7 @@ define('ext.wikia.adEngine.slotTweaker', [
 	'wikia.log',
 	'wikia.document',
 	'wikia.window'
-], function (log, document, window) {
+], function (log, doc, win) {
```
Comment (fraszczakszymon) [inline extensions/wikia/AdEngine/js/SlotTweaker.js:6]: :gem:

File: extensions/wikia/AdEngine/js/SlotTweaker.js [hunk 2]
```diff
@@ -92,26 +92,31 @@ define('ext.wikia.adEngine.slotTweaker', [
...
 			slot.className += ' ' + standardLeaderboardSizeClass;
 		}
 	}
 
 	// TODO: fix it, it's a hack!
 	function removeTopButtonIfNeeded(slotname) {
-		if (isLeaderboard(slotname) && !isStandardLeaderboardSize(slotname)) {
-			log('removing TOP_BUTTON_WIDE', 3, logGroup);
-			hide('TOP_BUTTON_WIDE');
+		function isEnabled() {
```
Comment (fraszczakszymon) [inline extensions/wikia/AdEngine/js/SlotTweaker.js]: Why not just a boolean flag?

File: extensions/wikia/AdEngine/js/SlotTweaker.js [hunk 3]
```diff
@@ -92,26 +92,31 @@ define('ext.wikia.adEngine.slotTweaker', [
...
 		if (isLeaderboard(slotname) && isStandardLeaderboardSize(slotname)) {
 			slot.className += ' ' + standardLeaderboardSizeClass;
 		}
 	}
 
 	// TODO: fix it, it's a hack!
 	function removeTopButtonIfNeeded(slotname) {
-		if (isLeaderboard(slotname) && !isStandardLeaderboardSize(slotname)) {
-			log('removing TOP_BUTTON_WIDE', 3, logGroup);
-			hide('TOP_BUTTON_WIDE');
```
Comment (fraszczakszymon) [inline extensions/wikia/AdEngine/js/SlotTweaker.js:105]: What about hiding it?

File: extensions/wikia/Spotlights/js/AdProviderOpenX.js [hunk 4]
```diff
@@ -55,19 +55,40 @@ AdProviderOpenX.getUrl = function() {
...
 		return window.Wikia.geo.isProperGeo(window.Wikia.InstantGlobals.wgReviveSpotlightsCountries);
 	} catch (e) {
 		return false;
 	}
 }
 
+function createReviveLazyQueue() {
+	'use strict';
+
+	window.Wikia.reviveQueue = window.Wikia.reviveQueue || [];
```
Comment (fraszczakszymon) [inline extensions/wikia/Spotlights/js/AdProviderOpenX.js:69]: What do you think about moving it to `window.ads.reviveSlots`?

File: extensions/wikia/WikiaBar/js/WikiaBar.js [hunk 5]
```diff
@@ -62,13 +61,21 @@ var WikiaBar = {
 		return true;
 	},
 	getAdIfNeeded: function () {
-		var WikiaBarBoxAd = $('#' + this.WIKIA_BAR_BOXAD_NAME),
-			showAds = window.ads && window.ads.context && window.ads.context.opts && window.ads.context.opts.showAds;
+		function isEnabled() {
```
Comment (fraszczakszymon) [inline extensions/wikia/WikiaBar/js/WikiaBar.js]: Why not just a boolean flag?

File: extensions/wikia/WikiaBar/js/WikiaBar.js [hunk 6]
```diff
@@ -62,13 +61,21 @@ var WikiaBar = {
...
+				window.wgEnableWikiaBarAds;
+		}
+
+		if (isEnabled()) {
+			window.Wikia.reviveQueue = window.Wikia.reviveQueue || [];
 
-		if( WikiaBarBoxAd.hasClass('wikia-ad') == false && showAds && window.wgEnableWikiaBarAds ) {
-			// TODO: refactor this once AdEngine calls you back on error/success
-			window.adslots2.push(this.WIKIA_BAR_BOXAD_NAME);
-			WikiaBarBoxAd.addClass('wikia-ad');
```
Comment (fraszczakszymon) [inline extensions/wikia/WikiaBar/js/WikiaBar.js]: What about `wikia-ad` class?

File: extensions/wikia/AdEngine/js/SlotTweaker.js (hunk 3 above)
Comment (drets) [inline extensions/wikia/AdEngine/js/SlotTweaker.js:105]: No need.

File: extensions/wikia/AdEngine/js/SlotTweaker.js (hunk 1 above)
Comment (drets) [inline extensions/wikia/AdEngine/js/SlotTweaker.js:6]: :bow:

**Comment by fraszczakszymon** [comment]
LGTM 👍

**Comment by fraszczakszymon** [comment]
LGTM 🙈

**Pull request merged**